rust-engine/src/backends/lean.rs

```diff
@@ -1707,3 +1728,32 @@ set_option linter.unusedVariables false
         }
     }
 };
+
```

Comment:
Maybe these tests can also go into `tests/literals/src/lib.rs`? Or maybe `tests/lean-tests` if they fail for the other backends?